module/x/gravity/keeper/keeper.go

```diff
@@ -464,6 +464,7 @@ func (k Keeper) SetOutgoingTx(ctx sdk.Context, outgoing types.OutgoingTx) {
 // DeleteOutgoingTx deletes a given outgoingtx
 func (k Keeper) DeleteOutgoingTx(ctx sdk.Context, storeIndex []byte) {
 	ctx.KVStore(k.storeKey).Delete(types.MakeOutgoingTxKey(storeIndex))
+	k.DeleteEthereumSignatures(ctx, storeIndex)
```

Comment:
I'm not certain if it matters in practice here, but I would suggest cleaning up the "child" records (the signatures) before deleting the "parent" record (the outgoing TX).